### ISSUE TITLE
Retain original feed's favicon

### ISSUE THREAD

**Issue by kriegaex** [spenibus/rss-filter-php#5]
I noticed that all my filtered feeds have the same favicon: my own website's. This seems logical and is just a cosmetic problem.

Before migrating to this self-hosted solution, however, I used an external, hosted feed filtering service. Somehow, that service managed to pass through the original feeds' icons. I am not an Atom or RSS expert, so I have no idea if and how it would be possible to add the same feature to this project too, but if so, I would like to see it implemented. If you know how it could be done and can explain it in layman's terms to me, I might even take a stab at it myself next time I have some free cycles.

**Comment by spenibus** [comment]
As a reference, when serving HTML from a website, there are 2 ways to do this, the gold old favicon.ico at the root (which would be quite obviously very static) or explicitly set an image through a link tag (allowing as many as you want in any supported file format):

    <!DOCTYPE html>
    <html>
       <head>
          <link rel="icon" type="image/png" href="/path/to/the/icon.png" />

I am however not aware of such a mechanism in RSS, it would require some investigation.

**Comment by kriegaex** [comment]
> the good old favicon.ico at the root (which would be quite obviously very static) or explicitly set an image through a link tag

Yes, thanks, I know that.

> I am however not aware of such a mechanism in RSS, it would require some investigation. Perhaps through the main [image tag](https://www.rssboard.org/rss-specification#ltimagegtSubelementOfLtchannelgt).

Yes, I found that information too. With Atom, it seems to be more flexible, there are separate `logo` and `icon` tags, see [this article](https://snook.ca/archives/rss/add_logo_to_feed). Now for me personally, the more important question is what InoReader does, because it is the tool I use on all devices. I do not use any locally installed client or built-in e-mail tool feed reader. I just sent a support question to the InoReader folks. I know that some reader services simply ignore the feed tags, only relying on favicon.ico. Google seems to be one such service, but at least it seems to follow redirects. So in that case, the only things that would help would be adding specific redirects for each feed to the webserver configuration or to host each feed on a subdomain URL and put a static favicon file to each subdomain. My hosting plan has lots of subdomains included, I could of course do that. But if the InoReader people reply that the feed tags are respected, I would rather patch extra information into `rss-filter-php`, e.g. reading a favicon URL from the XML configuration.

**Comment by kriegaex** [comment]
Before I am closing this issue, I want to document how I solved it using Apache web server configuration for my shared hosting provider:

### 1. Create one subdomain per filtered feed

Assuming, we create subdomains `first.feeds.mydomain.com` and `second.feeds.mydomain.com`:
* Make each subdomain point to the exact same  _rss-filter-php_ installation directory
* Optionally, create _Let's Encrypt_ SSL certificates for each subdomain in order to enable HTTPS. Because this can be done via mouse clicks in my provider's admin interface, I did that.
* Resulting feed URLs would be something like:
  * `https://first.feeds.mydomain.com/index.php?config=first`
  * `https://second.feeds.mydomain.com/index.php?config=second`

### 2. Use `.htaccess` to redirect favicon URLs

Upload a file like this to the _rss-filter-php_ installation directory:
```text
RewriteEngine on

RewriteCond "%{HTTP_HOST}" "first.feeds.mydomain.com"
RewriteRule "^favicon\.ico$" "https://first_feed_original_website.net/favicon.ico" [R,L]

RewriteCond "%{HTTP_HOST}" "second.feeds.mydomain.com"
RewriteRule "^favicon\.ico$" "https://second_feed_original_website.org/favicon.ico" [R,L]
 ```

### 3. Subscribe to subdomain feed URLs in your favourite feed reader

Now you should be able to enjoy the original favicons.
